contrib/win32/win32compat/shell-host.c

```diff
@@ -768,22 +817,14 @@ ProcessEvent(void *p)
 			return GetLastError();
 		}
 
-		SendBuffer(pipe_out, pBuffer, bufferSize);
+		SendBufferAtPos(pipe_out, wX, wY, pBuffer, bufferSize);
 		break;
 	}
 	case EVENT_CONSOLE_UPDATE_SCROLL:
 	{
-		DWORD out = 0;
-		LONG vd = idChild;
-		LONG hd = idObject;
-		LONG vn = abs(vd);
-
-		if (vd > 0) {
-			if (ViewPortY > 0)
-				ViewPortY -= vn;
-		} else {
-			ViewPortY += vn;
-		}
+		ViewPortY -= idChild;
+		if (ViewPortY > INT_MAX)
```

Comment:
why the existing code is removed? the scoll can be upwards or the downwards so we need to either add /delete as per existing code..